### ISSUE TITLE
StackOverflowError with Spoon 6.0.0 using the MavenLauncher

### ISSUE THREAD

**Issue by orichalque** [INRIA/spoon#1617]
Hi, I'm having some issues while spooning the source code of this project: https://github.com/joel-costigliola/assertj-core

The MavenLauncher was working really well so far, on other smaller projects, but I'm having this error with this project:

```
java.lang.StackOverflowError
	at spoon.support.visitor.clone.CloneBuilder.visitCtTypeReference(CloneBuilder.java:241)
	at spoon.reflect.visitor.CtInheritanceScanner.visitCtTypeParameterReference(CtInheritanceScanner.java:797)
	at spoon.support.visitor.clone.CloneBuilder.visitCtTypeParameterReference(CloneBuilder.java:236)
	at spoon.reflect.visitor.CtInheritanceScanner.visitCtWildcardReference(CtInheritanceScanner.java:802)
	at spoon.support.reflect.reference.CtWildcardReferenceImpl.accept(CtWildcardReferenceImpl.java:28)
	at spoon.reflect.visitor.CtInheritanceScanner.scan(CtInheritanceScanner.java:170)
	at spoon.support.visitor.clone.CloneBuilder.build(CloneBuilder.java:33)
	at spoon.support.visitor.clone.CloneVisitor.visitCtWildcardReference(CloneVisitor.java:646)
	at spoon.support.reflect.reference.CtWildcardReferenceImpl.accept(CtWildcardReferenceImpl.java:28)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.support.visitor.equals.CloneHelper.clone(CloneHelper.java:48)
	at spoon.support.visitor.equals.CloneHelper.clone(CloneHelper.java:72)
	at spoon.support.visitor.clone.CloneVisitor.visitCtTypeReference(CloneVisitor.java:670)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.accept(CtTypeReferenceImpl.java:83)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:150)
	at spoon.support.visitor.equals.CloneHelper.clone(CloneHelper.java:48)
	at spoon.support.reflect.declaration.CtElementImpl.clone(CtElementImpl.java:479)
	at spoon.support.reflect.reference.CtReferenceImpl.clone(CtReferenceImpl.java:79)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.clone(CtTypeReferenceImpl.java:803)
	at spoon.support.visitor.AbstractTypingContext.adaptType(AbstractTypingContext.java:58)
	at spoon.support.visitor.MethodTypingContext.adaptType(MethodTypingContext.java:43)
...
```

My configuration is the following: 

```
            Launcher launcher = new MavenLauncher(path, MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
            launcher.getEnvironment().setLevel(String.valueOf(Level.ALL));
            launcher.getEnvironment().setShouldCompile(true);
            launcher.getEnvironment().setAutoImports(true);
            launcher.getEnvironment().setNoClasspath(false);

            launcher.run();
```

I made a small maven project producing this errror when running a mvn clean install through IntelliJ IDEA. 
[SpoonStackError.zip](https://github.com/INRIA/spoon/files/1397738/SpoonAssertJ.zip)

Any idea from where this might come from ? 

Thank you :+1:

**Comment by tdurieux** [comment]
The issue comes from our AutoImports implementation.
I did not yet found the origin of the error.

Thanks for the report.

**Comment by orichalque** [comment]
Alright, I'll try this out then.
Thank you !

**Comment by surli** [comment]
I just have a look, I'll propose a PR for fixing the bug. 
But, you'll have some issues running this project because some dependencies are marked as provided in the pom.xml and Spoon cannot handle such dependencies: so you have to edit the environment classpath to add yourself the missing dependencies.

**Comment by surli** [comment]
@orichalque I did not manage to reproduce the bug in a smaller test case, and it's possible it has been fixed with changes made in #1672 
I close it for now, don't hesitate to comment it back or to open a new issue if the problem is still there.
